Web_Development/Javascript/Git and Bash/readme.md

```diff
@@ -0,0 +1,139 @@
+- For `Mac` users , Git already comes installed with their `MacOS`. you can check whether it is installed not by opening up your terminal and type.
+```shell
+git -version
+```
+- In maximum `Linux` distro git comes with already installed  but you can install it but depends upon which linux distro and package manager you are using.
+
+##### For Ubuntu
+```shell
+sudo apt install git
+```
+##### For Arch based distros
+```shell
+sudo pacman -Syu git
+```
+### Basic commands in git
+
+-  initializing a directory with git.
+```shell
+git init 
+```
+- Check logs of commits of current branch
+```shell
+git log
+```
+- Check whether a new changes are occured or not
+```shell
+git status
+```
+- make a new commit 
+```shell
+git add . && git commit -m "your message"
+```
+    make sure your message will be some meaning full text
+	you can also use a short command for this
+```shell
+git commit -am "your message"
+```
+- switch to  a particular commit
+```shell
+git checkout commit_id
+```
+    `commit_id` are the alphanumeric number which you get while `git logs`
+	
+- Making a new branch 
+```shell
+git branch branch_name
+```
+- shifting to a particular branch
+```shell
+git checkout branch_name
+```
+- deleting a branch
+```shell
+git branch -d branch_name
+```
+These are some basic commands which you will use for basic git operations.
+<hr>
+
+### Bash
+Bash also known <b>Bourne Again Shell</b> is a standard shell which available in Unix based system as a default shell. Nowadays it is comes as a standard shell in all GNU/LINUX system. Here term `shell` is outmost layer of kernal. so shell do the task of interacting with kernal by user commands.
+
+### Features of Bash
+- You can perform not only just one line commands but even multiline commands so basically you can use it as programming language.
+- Bash is suitable for using scripting related tasks.
+- Bash gives you interactive environment for executing any commands..
+- Bash is very customisable and flexible You can even make your configuration in `.bashrc` file which is situated in user `$HOME` directory.
+- Bash is more secure than any other shell . You can not perform any execution of any type of file untill user does not give permission to make that file executable.
+
+These are some basic features of Bash . 
+
+### Basic commands in Bash
+- echo or print any text
+```shell
+echo your_text
+```
+- listing out all directories and files
+```shell
+ls
+```
+- creating a new directory
+```shell
+mkdir directory_name
+```
+- creating a file
+```shell
+touch file_name
+```
+- finding pattern in a text file
+```shell
+grep `word_to_find`
+```
+- you can pipe symbol to treat output of a some commands as an input for some command.
+```shell
+echo this is me | grep me
+```
+- removing file and directory
+```shell
+//This will only remove file
+rm file_name 
+```
+```shell
+//This will remove an empty directory
+rm -f directory_name
+```
+```shell
+//This will remove non empty directory
```

Comment:
For bash, instead of using `//` use `#` for showing comments in the code.

Reply:
Please do the requested change here.

Reply:
i have fixed this in new commits please recheck.